Sources/TuistKit/Generator/Generator.swift

```diff
@@ -79,20 +83,24 @@ class Generator: Generating {
     }
 
     func load(path: AbsolutePath) async throws -> (Graph, [SideEffectDescriptor]) {
-        try await manifestGraphLoader.load(path: path)
+        let (graph, sideEffectDescriptors, issues) = try await manifestGraphLoader.load(path: path)
+        lintWarningIssues.append(contentsOf: issues)
+        return (graph, sideEffectDescriptors)
     }
 
     private func lint(graphTraverser: GraphTraversing) throws {
         let config = try configLoader.loadConfig(path: graphTraverser.path)
+        let environmentIssues = try environmentLinter.lint(config: config)
+        lintWarningIssues.append(contentsOf: environmentIssues.printAndThrowIfNeeded())
 
-        try environmentLinter.lint(config: config).printAndThrowIfNeeded()
-        try graphLinter.lint(graphTraverser: graphTraverser).printAndThrowIfNeeded()
+        let graphIssues = try graphLinter.lint(graphTraverser: graphTraverser).printErrorsAndThrowIfNeeded()
```

Comment:
The print helper function is doing a few operations 😅 

- Printing
- Throwing
- Filtering and returning any warnings


What do you think about decomposing it slightly - e.g. something like

```swift
    private func lint(graphTraverser: GraphTraversing) throws {
         let config = try configLoader.loadConfig(path: graphTraverser.path)

         let environmentIssues = try environmentLinter.lint(config: config)
         try environmentIssues.printAndThrowErrorsIfNeeded()
         lintWarningIssues.append(contentsOf: environmentIssues)

         let graphIssues = try graphLinter.lint(graphTraverser: graphTraverser)
         try graphIssues.printAndThrowErrorsIfNeeded()
         lintWarningIssues.append(contentsOf: issues)
    }
```